docs/releasing.md

```diff
@@ -4,102 +4,40 @@ Describes how we do releases of Mithril.js
+- Publishing releases and pre-releases to npm
+- Creating github releases
+- Rollbacks
 
-### Prepare the release
+## For contributors
 
-1. Ensure your local branch is up to date
+Contributors should create their feature branch targetting the default branch `next`.  When this branch is merged `pr-release` will either generate or update a release PR from `next` to `main`.
 
-```bash
-$ git checkout next
-$ git pull --rebase mithriljs next
-```
-
-2. Determine patch level of the change
-3. Update information in `docs/changelog.md` to match reality of the new version being prepared for release.
-	- Don't forget to add today's date under the version heading!
-4. Replace all existing references to `mithril@next` to `mithril` if moving from a release candidate to stable.
-    - Note: if making an initial release candidate, don't forget to move all the playground snippets to pull from `mithril@next`!
-5. Commit changes to `next`
-
-```
-$ git add .
-$ git commit -m "Preparing for release"
-
-# Push to your branch
-$ git push
-
-# Push to MithrilJS/mithril.js
-$ git push mithriljs next
-```
-
-### Merge from `next` to `master`
-
-6. Switch to `master` and make sure it's up to date
-
-```bash
-$ git checkout master
-$ git pull --rebase mithriljs master
-```
-
-7. merge `next` on top of it
-
-```bash
-$ git merge next
-```
+The description and title will be managed by [pr-release], including the semver version.
 
-8. Clean & update npm dependencies and ensure the tests are passing.
+Contributors who have permissions should add the correct semver label to their PR (`major` | `minor` | `patch`).  If no label is set, `patch` is assumed.
 
-```bash
-$ npm prune
-$ npm i
-$ npm test
-```
-
-### Publish the release
+If you do not have permissions, the maintainer will set the label on your behalf.
 
-9. `npm run release <major|minor|patch|semver>`, see the docs for [`npm version`](https://docs.npmjs.com/cli/version)
-10. The changes will be automatically pushed to your fork
-11. Push the changes to `MithrilJS/mithril.js`
-
-```bash
-$ git push mithriljs master
-```
+## Changelog
 
-12. Travis will push the new release to npm & create a GitHub release
+There are two changelogs in the Mithril.js project
 
-### Merge `master` back into `next`
+- `docs/changelog.md` a hand written curated reflection of changes to the codebase
+- `docs/release.md` an automatically prepended log of changes, managed by pr-release
 
-This helps to ensure that the `version` field of `package.json` doesn't get out of date.
+In future we may collapse these into a single file, the separation is due to the fact the `changelog.md` predates the `release.md` file.
 
-13. Switch to `next` and make sure it's up to date
+## For maintainers
 
-```bash
-$ git checkout next
-$ git pull --rebase mithriljs next
-```
+Whenever a new feature branch is opened, a reviewing maintainer should add the correct semver label to their PR (`major` | `minor` | `patch`).  If no label is set, `patch` is assumed.
```

Comment:
What are your thoughts on always requiring a label to be set for non-Dependabot merges?

Reply:
I would vote for it. Is there a way to force one of the labels for a pr?

Reply:
@JAForbes Could I get your word on this? I'll approve once this immediate discussion here is either resolved or moved to its own issue or something.

Reply:
I think requiring a label isn't really in scope for this PR, feels like a good feature idea though but would require some thought / discussion.

Reply:
Okay, approved. Could you create an issue/discussion so we can elaborate on that further? Don't want it to get lost, because it's kinda important to avoid a pretty major potential future mishap.